tests/unit/test_ntm.py

```diff
@@ -0,0 +1,288 @@
+
+
+def test_num_patience_epochs_validation_fail_type(sagemaker_session):
+    with pytest.raises(ValueError):
+        NTM(num_patience_epochs='other', sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+def test_num_patience_epochs_validation_fail_value(sagemaker_session):
+    with pytest.raises(ValueError):
+        NTM(num_patience_epochs=0, sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+def test_rescale_gradient_fail_type(sagemaker_session):
+    with pytest.raises(ValueError):
+        NTM(rescale_gradient='other', sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+def test_rescale_gradient_validation_fail_value(sagemaker_session):
+    with pytest.raises(ValueError):
+        NTM(rescale_gradient=0, sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+def test_clip_gradient_fail_type(sagemaker_session):
+    with pytest.raises(ValueError):
+        NTM(clip_gradient='other', sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+def test_clip_gradient_validation_fail_value(sagemaker_session):
+    with pytest.raises(ValueError):
+        NTM(clip_gradient=0, sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+def test_weight_decay_fail_type(sagemaker_session):
+    with pytest.raises(ValueError):
+        NTM(weight_decay='other', sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+def test_weight_decay_validation_fail_value(sagemaker_session):
+    with pytest.raises(ValueError):
+        NTM(weight_decay=2, sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+def test_learning_rate_fail_type(sagemaker_session):
+    with pytest.raises(ValueError):
+        NTM(learning_rate='other', sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+def test_learning_rate_validation_fail_value(sagemaker_session):
+    with pytest.raises(ValueError):
+        NTM(learning_rate=2, sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+PREFIX = "prefix"
+BASE_TRAIN_CALL = {
+    'hyperparameters': {},
+    'image': registry(REGION, "ntm") + '/ntm:1',
+    'input_config': [{
+        'DataSource': {
+            'S3DataSource': {
+                'S3DataDistributionType': 'ShardedByS3Key',
+                'S3DataType': 'ManifestFile',
+                'S3Uri': 's3://{}/{}'.format(BUCKET_NAME, PREFIX)
+            }
+        },
+        'ChannelName': 'train'
+    }],
+    'input_mode': 'File',
+    'output_config': {'S3OutputPath': 's3://{}/'.format(BUCKET_NAME)},
+    'resource_config': {
+        'InstanceCount': TRAIN_INSTANCE_COUNT,
+        'InstanceType': TRAIN_INSTANCE_TYPE,
+        'VolumeSizeInGB': 30
+    },
+    'stop_condition': {'MaxRuntimeInSeconds': 86400}
+}
+
+FEATURE_DIM = 10
+MINI_BATCH_SIZE = 200
+HYPERPARAMS = {'num_topics': NUM_TOPICS, 'feature_dim': FEATURE_DIM, 'mini_batch_size': MINI_BATCH_SIZE}
+STRINGIFIED_HYPERPARAMS = dict([(x, str(y)) for x, y in HYPERPARAMS.items()])
+HP_TRAIN_CALL = dict(BASE_TRAIN_CALL)
```

Comment:
If this is not being used anywhere please remove.

Reply:
Removed unnecessary parameters.